### PR TITLE
Remove the spark-sample-app from the otel-test-framework

### PR DESCRIPTION
**Pull request by vasireddy99**
This PR addresses the issue #497 

In-order to reduce the maintenance for [spark-sample app](https://github.com/aws-observability/aws-otel-java-instrumentation/tree/main/sample-apps/spark) form aws-otel-java-instrumentation in aws-otel-test-framework. Using the public ECR image `public.ecr.aws/aws-otel-test/aws-otel-java-spark:latest` as `sample-app-image` instead of dedupe.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

**Review by khanhntd — APPROVED** [review]
So the whole point would be transferring all the sample apps to their corresponding aws-otel SDK. Therefore, reducing maintenance the sample app on aws-otel-test-framework correct? If that's the whole point, then it's good to me

File: terraform/testcases/otlp_mock/parameters.tfvars [hunk 1]
```diff
@@ -1,2 +1,5 @@
 # data type will be emitted. Possible values: metric or trace
 soaking_data_mode = "trace"
+
+#sample app image
+sample_app_image = "public.ecr.aws/aws-otel-test/aws-otel-java-spark:latest"
```
Comment (jefchien) [inline terraform/testcases/otlp_mock/parameters.tfvars:7]: You'll want to change it in more places than this one. There are a lot of testcases (`logzio_exporter_trace_mock`, `newrelic_exporter_metric_mock`, etc.) that use the default sample app, which is the spark app.

**Review by jefchien — CHANGES_REQUESTED** [review]

**Comment by sethAmazon** [comment]
Does the spark latest support arm64 docker images?

**Comment by vasireddy99** [comment]
> Does the spark latest support arm64 docker images?

yes, it supports them [PR](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/140)

**Comment by jefchien** [comment]
@vasireddy99 It looks like it's meant to, but the image on ECR https://gallery.ecr.aws/aws-otel-test/aws-otel-java-spark only has one.

**Comment by vasireddy99** [comment]
Changing to draft till #526 is merged.

- `sample_app_image` takes precedence over `sample_app` couldn't take the approach of using it as default image

**Comment by vasireddy99** [comment]
Closed due to rebasing issues. created [#527](https://github.com/aws-observability/aws-otel-test-framework/pull/527/files)

**Pull request closed without merging**